pubsub/flow_controller.go

```diff
@@ -176,3 +201,25 @@ func (f *flowController) bound(size int) int64 {
 func (f *flowController) count() int {
 	return int(atomic.LoadInt64(&f.countRemaining))
 }
+
+func (f *flowController) recordOutstandingMessages(ctx context.Context, n int64) {
+	switch f.purpose {
+	case flowControllerPurposeTopic:
+		recordStat(ctx, PublisherOutstandingMessages, n)
+	case flowControllerPurposeSubscription:
+		fallthrough
```

Comment:
Made this fall through to maintain backwards compatibility with the old behavior. However, as written there shouldn't be any cases where a "real" client has an unknown purpose, so I'm happy to make `default` a noop. Or I can remove the unknown value entirely and default to `flowControllerPurposeSubscription`.

Reply:
I think we should remove the unknown value since flowcontroller is only used for publisher/subscribers here. Thanks!